### PR TITLE
Remove some Page ASM patches

### PR DESCRIPTION
**Pull request by GnomedDev**

### CODE REVIEW

File: payload/game/ui/AwardPage.cc [hunk 1]
```diff
@@ -55,6 +60,57 @@ void AwardPage::onInit() {
...
+        auto battleType = awardsScenario.battleType;
+        if (battleType == 0) {
+            messageId = 0x587;
+            srcPane = "icon_09_balloon";
+        } else if (battleType == 1) {
+            messageId = 0x588;
+            srcPane = "icon_10_coin";
+        }
+    } else {
+        panic("Unknown section 0x%X, please report to MKW-SP devs!", static_cast<s32>(currentSectionId));
```
Comment (riidefi) [inline payload/game/ui/AwardPage.cc:107]: What's the game's behavior here?

Comment (GnomedDev) [reply payload/game/ui/AwardPage.cc:107]: Stock treats sections it doesn't know about as VS (stock logic is a bit reordered and more repetitive). This should only possibly be hit for 0x37 which is marked as unknown and we do not transition to.

**Comment by GnomedDev** [comment]
This is okay to merge without, until the format the world PR is merged.

File: payload/game/ui/page/DemoPage.hh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "game/ui/GlobalContext.hh"
+#include "game/ui/Page.hh"
+
+namespace UI {
+
+class DemoPage : public Page {
+private:
+    REPLACE void onInit() override;
```
Comment (stblr) [inline payload/game/ui/page/DemoPage.hh:10]: public

File: payload/game/ui/TitlePage.hh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+#include "game/ui/Page.hh"
+
+namespace UI {
+
+class TitlePage : public Page {
+private:
+    void REPLACED(onInit)();
+    void REPLACED(onActivate)();
+    REPLACE void onInit() override;
+    REPLACE void onActivate() override;
```
Comment (stblr) [inline payload/game/ui/TitlePage.hh:12]: public

File: payload/game/ui/page/DemoPage.cc [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        childCount += globalContext->m_matchCount * 2;
+    }
+
+    initChildren(childCount);
+    insertChild(0, &m_courseDisplay, 0);
+    insertChild(1, &m_cupDisplay, 0);
+
+    ControlLoader courseDisplayLoader(&m_courseDisplay);
+    ControlLoader cupDisplayLoader(&m_cupDisplay);
+
```
Comment (stblr) [inline payload/game/ui/page/DemoPage.cc:35]: Please use `LayoutUIControl::load`.

File: payload/game/ui/ControlLoader.hh
Comment (stblr) [inline payload/game/ui/ControlLoader.hh:1]: Can be axed.

**Review by stblr — CHANGES_REQUESTED** [review]

File: payload/game/ui/ControlLoader.hh
Comment (stblr) [reply payload/game/ui/ControlLoader.hh:1]: Or be used to implement `LayoutUIControl::load` in C++.

File: payload/game/ui/page/DemoPage.cc (hunk 4 above)
Comment (GnomedDev) [reply payload/game/ui/page/DemoPage.cc:35]: Huh, I don't see that in the original game, was it inlined or is it an SP thing?

Comment (stblr) [reply payload/game/ui/page/DemoPage.cc:35]: Inlined.

Comment (stblr) [reply payload/game/ui/page/DemoPage.cc:35]: https://github.com/stblr/mkw-sp/blob/4aaf4b0b374658f4eaeabe31b57a97d567188523/payload/game/ui/UIControl.c#L14

Comment (GnomedDev) [reply payload/game/ui/page/DemoPage.cc:35]: Okay, that's going to be my next target for cleanup, for sure.

**Review by stblr — APPROVED** [review]

**Pull request merged**